R/motif_enrichment_HOMER.R

```diff
@@ -95,6 +96,16 @@ dumpJaspar <- function(filename,
     # load PFMs and convert to HOMER format (base probabilties)
     requireNamespace(pkg)
     mdb <- utils::getFromNamespace(pkg, ns = pkg)
+    if (!methods::existsMethod(f = getMatrixSet, signature = list(x = class(mdb)))) {
+        # workaround for JASPAR2024 (no specific TFBSTools::getMatrixSet method yet)
+        if (inherits(mdb, "function") && exists(x = "db", where = paste0("package:", pkg))) {
+            jobj <- do.call(what = mdb, args = list())
+            dbfunc <- get(x = "db", envir = as.environment(paste0("package:", pkg)))
+            mdb <- RSQLite::dbConnect(RSQLite::SQLite(), dbfunc(jobj))
```

Comment:
Should we also add an explicit `dbDisconnect()`?

Reply:
Good point. I have added `on.exit(expr = RSQLite::dbDisconnect(mdb))`, or do you think it would be better to do it using an if-clause at the end?

Reply:
I think `on.exit()` is good!